prototype/core/impl/src/main/java/org/eclipse/sensinact/prototype/command/impl/GatewayThreadImpl.java

```diff
@@ -155,7 +164,15 @@ public void run() {
             try {
                 WorkItem<?> item = work.take();
                 currentItem.set(item);
-                item.doWork();
+
+                metrics.getCounter("sensinact.tasks.pending").dec();
+                metrics.getHistogram("sensinact.tasks.pending.hist").update(work.size());
+                try (IMetricTimer timer = metrics.withTimer("sensinact.task.time")) {
+                    item.doWork();
+                } catch (Exception e) {
+                    // Ignore: the timer shouldn't fail on close
+                    e.printStackTrace();
```

Comment:
This will break the behaviour where `item.doWork()` throws an `InterruptedException`

The Timer (and other interfaces) should override the close method to remove the exception as described in https://stackoverflow.com/questions/25959779/why-close-method-of-java-lang-autocloseable-throws-exception-but-close-method-o#answer-49515309 - this will allow try-with-resources, but it will never need a `catch (Exception)` block